### PR TITLE
Fix oh-my-fish

### LINKED ISSUE #513: Could you please add oh-my-fish?
**I want to suggest a new step**
***Which tool is this about? Where is its repository?***
- oh-my-fish
- The Fish Shell Framework
- https://github.com/oh-my-fish/oh-my-fish
***Which operating systems are supported by this tool?***
- Linux/Mac/WSL
***What should Topgrade do to figure out if the tool needs to be invoked?***
- `omf update`
***Which exact commands should Topgrade run?***
- None

### PR DESCRIPTION
**Pull request by niheaven**
Fix #513, just like `fisher`.

## Standards checklist:

- [x] The PR title is descriptive.
- [x] The code compiles
- [x] The code passes rustfmt
- [x] The code passes clippy
- [x] The code passes tests
- [X] *Optional:* I have tested the code myself
    - [x] I also tested that Topgrade skips the step where needed

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -240,7 +240,7 @@ fn run() -> Result<()> {
         runner.execute(Step::Shell, "zinit", || zsh::run_zinit(&base_dirs, run_type))?;
         runner.execute(Step::Shell, "oh-my-zsh", || zsh::run_oh_my_zsh(&ctx))?;
         runner.execute(Step::Shell, "fisher", || unix::run_fisher(&base_dirs, run_type))?;
-        runner.execute(Step::Shell, "oh-my-fish", || unix::run_oh_my_fish(&ctx))?;
+        runner.execute(Step::Shell, "oh-my-fish", || unix::run_oh_my_fish(&base_dirs, run_type))?;
```
Comment (r-darwish) [inline src/main.rs:243]: Passing `&ctx` is the preferred way now. Can you please do that?

**Review by r-darwish — APPROVED** [review]

**Review by r-darwish — CHANGES_REQUESTED** [review]
Please pass `&ctx` to fish

File: src/main.rs (hunk 1 above)
Comment (niheaven) [reply src/main.rs:243]: So, could you please tell me how to pass `&base_dirs` with `&ctx`? I'm unfamiliar with Rust...

Or maybe `run_oh_my_fish(&base_dirs, &ctx)`?

Comment (r-darwish) [reply src/main.rs:243]: `ctx.base_dirs()` to access the base directories

Comment (niheaven) [reply src/main.rs]: Done.

**Review by r-darwish — APPROVED** [review]

**Pull request merged**